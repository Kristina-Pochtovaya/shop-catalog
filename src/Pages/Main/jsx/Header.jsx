import React from 'react';
import {Link} from 'react-router-dom';
import logo from '../img/logo.png'

export const Header = ({linkItem, link}) => (
  <div className="header-wrap">
    <div className="header-box">
      <div className="logo"> 	
        <img src={logo} alt="Все для дома" title="Все для дома"/>
      </div>
      <Link to={link} >
        {linkItem}
      </Link>
    </div>
  </div>
);



