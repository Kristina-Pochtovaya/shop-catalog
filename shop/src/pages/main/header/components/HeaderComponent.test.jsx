import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Header from './HeaderComponent';
import aboutUS from '../../../../assets/header/about-us.png';

configure({ adapter: new Adapter() });

const pages = {
  loginPersonalAccountReducer: {
    loginFormIsVisible: true,
  },
};

const myMethod = jest.fn().mockImplementation(() => true, true, false);

test('should render Header Component', () => {
  const header = shallow(<Header
    linkItem={<button type="button" className="buttonBack">Назад</button>}
    link="/main-page"
    disabled
    pages={pages}
    onLogin={myMethod}
  />);

  expect(toJson(header)).toMatchSnapshot();
});

test('if componenent has props with element button, it renders it', () => {
  const header = shallow(<Header
    linkItem={<button type="button" className="buttonBack">Назад</button>}
    link="/main-page"
    pages={pages}
    onLogin={myMethod}
  />);

  expect(header.find('button')).toHaveLength(1);
});

test('if componenent has props with img element, it renders it', () => {
  const header = shallow(<Header
    linkItem={<img src={aboutUS} alt="О нас" />}
    link="/about"
    pages={pages}
    onLogin={myMethod}
  />);

  expect(header.find('img')).toHaveLength(3);
  expect(header.find('button')).toHaveLength(0);
});
