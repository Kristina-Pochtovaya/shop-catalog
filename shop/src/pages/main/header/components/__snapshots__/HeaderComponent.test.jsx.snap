// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header Component 1`] = `
<div
  className="header-wrap"
>
  <div
    className="header-box"
  >
    <div
      className="logo"
    >
      <img
        alt="Все для дома"
        src="logo.png"
        title="Все для дома"
      />
    </div>
    <div
      className="basket -disabled"
    >
      <Link
        to="/basket"
      >
        <img
          alt="Корзина"
          src="basket.png"
          title="Корзина"
        />
      </Link>
    </div>
    <div
      className="headerNavigationButton"
    >
      <Link
        to="/main-page"
      >
        <button
          className="buttonBack"
          type="button"
        >
          Назад
        </button>
      </Link>
    </div>
  </div>
</div>
`;
